### PR TITLE
[ADD] delivery_carrier_b2c

### PR DESCRIPTION
**Pull request by bealdav**
Add specific fields to partner for deliveries purpose.

Some carriers allow to send goods directly to the final customer address
Some fields like intercom, door_code are required

Used for ColiPoste FR - Colissimo / So Colissimo

### CODE REVIEW

File: delivery_carrier_b2c/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from . import partner   # noqa
```
Comment (gurneyalex) [inline delivery_carrier_b2c/__init__.py]: please remove noqa comment

File: delivery_carrier_b2c/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        'base_delivery_carrier_label',
+        'base_partner_helper',
+    ],
+    'description': """
+Delivery Carrier Business To Customer
+=====================================
+
+Description
+-----------
+* Add specific fields to partner for deliveries purpose
```
Comment (gurneyalex) [inline delivery_carrier_b2c/__openerp__.py]: s/deliveries/delivery/

File: delivery_carrier_b2c/partner.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        'door_code': fields.char(
+            'Door Code',
+            size=5,),
+        'door_code2': fields.char(
+            'Door Code 2',
+            size=5),
+        'intercom': fields.char(
+            'Intercom',
+            size=32,
+            help="Informations for Intercom as name or number on the intercom"),
```
Comment (gurneyalex) [inline delivery_carrier_b2c/partner.py]: s/as/such as/

File: delivery_carrier_b2c/partner.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    _columns = {
+        'use_b2c_info': fields.boolean(
+            'Advanced address',
+            help="Display additional information for home delivery (b2c)"),
+        'door_code': fields.char(
+            'Door Code',
+            size=5,),
+        'door_code2': fields.char(
+            'Door Code 2',
+            size=5),
```
Comment (gurneyalex) [inline delivery_carrier_b2c/partner.py]: the size limits seem arbitrary. I know of places where the door code is longer than that...

**Comment by gurneyalex** [comment]
please add missing .pot file

**Comment by bealdav** [comment]
delivery_carrier_b2c module fails because require base_partner_helper module
latter module is defined here https://github.com/OCA/partner-contact/pull/29

**Comment by yvaucher** [comment]
:+1:

**Comment by achaussende** [comment]
:+1:

**Pull request merged**